digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/TrafficSignService.scala

```diff
@@ -410,55 +225,26 @@ class TrafficSignService(val roadLinkService: RoadLinkService, val userProvider:
     GeometryUtils.calculateActualBearing(validityDirection, Some(linkBearing)).get
   }
 
-  private def generateProperties(tRTrafficSignType: TRTrafficSignType, value: Any, additionalInfo: String) = {
-    val signValue = value.toString
-    val signAdditionalInfo = additionalInfo
-    val trafficType = tRTrafficSignType.trafficSignType
-    val typeProperty = SimpleProperty(typePublicId, Seq(PropertyValue(trafficType.value.toString)))
-    val valueProperty = additionalInfoTypeGroups.exists(group => group == trafficType.group) match {
-      case true => SimpleProperty(infoPublicId, Seq(PropertyValue(signAdditionalInfo)))
-      case _ => SimpleProperty(valuePublicId, Seq(PropertyValue(signValue)))
-    }
-
-    Set(typeProperty, valueProperty)
-  }
-
-  def createFromCoordinates(lon: Long, lat: Long, trafficSignType: TRTrafficSignType, value: Option[Int],
-                            twoSided: Option[Boolean], trafficDirection: TrafficDirection, bearing: Option[Int],
-                            additionalInfo: Option[String], roadLinks: Seq[VVHRoadlink]): Long = {
-
-      val closestLink: VVHRoadlink = roadLinks.minBy(r => GeometryUtils.minimumDistance(Point(lon, lat), r.geometry))
-      val (vvhRoad, municipality) = (roadLinks.filter(_.administrativeClass != State), closestLink.municipalityCode)
+  def createFromCoordinates(trafficSign: IncomingTrafficSign, closestLink: RoadLink, nearbyLinks: Seq[VVHRoadlink]): Long = {
 
-      if (vvhRoad.isEmpty || vvhRoad.size > 1) {
-        val asset = IncomingTrafficSign(lon, lat, 0, generateProperties(trafficSignType, value.getOrElse(""), additionalInfo.getOrElse("")), SideCode.Unknown.value, None)
-        createFloatingWithoutTransaction(asset, userProvider.getCurrentUser().username, municipality)
+    val (vvhRoad, municipality) = (nearbyLinks.filter(_.administrativeClass != State), closestLink.municipalityCode)
+    if (vvhRoad.isEmpty || vvhRoad.size > 1)
+      createFloatingWithoutTransaction(trafficSign.copy(linkId = 0), userProvider.getCurrentUser().username, municipality)
```

Comment:
Should a floating traffic sign be assigned bearing 'None', or the bearing of the closest road link?

Reply:
Could be None, if we use the linkId=0